lib/sdr_client/credentials.rb

```diff
@@ -12,6 +12,7 @@ def self.write(body)
       File.open(credentials_file, 'w', 0o600) do |file|
         file.flock(File::LOCK_EX)
         file.write(json.fetch('token'))
+        file.flush
```

Comment:
why do you need to flush if you're going to close immediately after this? Doesn't that do a flush?

Reply:
I would have thought so. However, still encountering problem reading file even with lock which suggests a flushing problem.

Ruby's [own example](https://ruby-doc.org/core-2.7.1/File.html#method-i-flock) shows flushing:
```
File.open("counter", File::RDWR|File::CREAT, 0644) {|f|
  f.flock(File::LOCK_EX)
  value = f.read.to_i + 1
  f.rewind
  f.write("#{value}\n")
  f.flush
  f.truncate(f.pos)
}
```

Reply:
But they aren't flushing right before a close. they are flushing before a truncate.